sim/vcf_sample.py

```diff
@@ -0,0 +1,81 @@
+import math
+import os
+import errno
+
+from utils import run
+
+from sample import Sample
+from genome import SimulatedGenome
+import config
+
+class VcfSample(Sample):
+    """ A sample generated from vcf file using simuG """
+    def __init__(self, 
+        predef_snp_path, 
+        reference_path=config.DEFAULT_REFERENCE_PATH,
+        seed=1, 
+        per_base_error_rate="0",
+        num_read_pairs=config.DEFAULT_NUM_READ_PAIRS,
+    ):
+
+        # TODO: Validate
+        self.reference_path = reference_path
+        
+        self.seed = seed
+        self.per_base_error_rate = per_base_error_rate
+        self.num_read_pairs = math.ceil(num_read_pairs)
+        
+        if os.path.isfile(predef_snp_path):
+            self.predef_snp_path = predef_snp_path
+        else:
+            raise Exception("Cant find the pre-defined snp path")
+        
+    @property
+    def name(self):
+        # extract filename
+        basename = os.path.splitext(os.path.basename(self.predef_snp_path))[0]
+        #TODO: this pattern is getting somewhat unweildy.
+        #   etiher use something more general or save metadata separately
+        return f"{type(self).__name__}-{basename}-seed{self.seed}-error{self.per_base_error_rate}-readpairs-{self.num_read_pairs}"
+
+    def decompose_complex_snps(self, output_file_path):
+        run(['vt', 
+            'decompose_blocksub',
+            self.predef_snp_path,
+            '-o', output_file_path])
+
+    def simulate_genome(self, simulated_genome_path, seed=1):
+        """ Simulated a genome with random SNPs
+
+            TODO: rename simulated_genome_path to simulated_genome_prefix
+
+            Parameters:
+                reference_path (str): Path to reference genome
+                simulated_genome_path (str): Path to simlated genome
+                predef_snps_path (str): Path to snippy generated VCF file.
+                seed (int): Seed value for simulation
+
+            Returns:
+                None
+        """
+        if not os.path.isfile(self.predef_snp_path):
+            raise(FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), self.predef_snp_path))
+        tmp_decomposed_vcf_path = simulated_genome_path + '.decomposed.vcf'
```

Comment:
I think we should consider having a "temp/" or "work/" directory at the base of the repo for putting this file (and any other temp files created) into, which could either be always deleted or only when running in light_mode. Just becuase it doesn't make sense to me to put it in `simulated_genome_path` even if it is temporary.

Reply:
Could be  TODO

Reply:
yeah good idea, TODO added. python has functionality for temp directories actually, that automatically clean themselves up. https://docs.python.org/3/library/tempfile.html